eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/PoolReap.hs

```diff
@@ -128,9 +118,12 @@ instance
   assertions =
     [ PostCondition
         "Deposit pot must equal obligation (PoolReap)"
-        ( \(TRC (env, _, _)) st ->
+        ( \_trc st ->
             potEqualsObligation
-              (mkCertState (speVState env) (prPState st) (prDState st))
+              ( def
+                  & certPStateL .~ prPState st
+                  & certDStateL .~ prDState st
```

Comment:
I am more than sure this approach will result in assertion violations. So, I would suggest instead is to replace `prDState` and `prPState` with `prCertState :: CertState era` and then this assertion implementation will be easy